upup/pkg/fi/cloudup/loader.go

```diff
@@ -214,6 +217,40 @@ func (l *Loader) addAssetCopyTasks(assets []*assets.ContainerAsset, lifecycle *f
 
 		}
 	}
+
+	return nil
+}
+
+func (l *Loader) addAssetFileCopyTasks(assets []*assets.FileAsset, lifecycle *fi.Lifecycle) error {
+	for _, asset := range assets {
+		if asset.CanonicalFileURLLocation != nil && asset.FileURL != asset.CanonicalFileURLLocation {
+			glog.V(10).Infof("processing asset: %q, %q", asset.FileURL.String(), asset.CanonicalFileURLLocation.String())
+			context := &fi.ModelBuilderContext{
+				Tasks: l.tasks,
+			}
+
+			glog.V(10).Infof("adding task: %q", asset.FileURL.String())
+			copyFileTask := &assettasks.CopyFile{
+				Name:       fi.String(asset.CanonicalFileURLLocation.String()),
```

Comment:
@justinsb you mentioned that you think we could reduce the number of struct members?